examples/toolchains/cc/hello.cc

```diff
@@ -0,0 +1,6 @@
+#include <iostream>
+
+int main()
+{
+	std::cout << "Hello world!\n";
```

Comment:
Can you please add a dependency to this program, so we know how that would look like?

Reply:
There are several ways in which a Bazel target could depend on a binary target. E.g. a [`tool` dependency on a `genrule`](https://docs.bazel.build/versions/main/be/general.html#genrule_examples), or as a [`data` dependency on a binary or a test](https://github.com/bazelbuild/codelabs/blob/f07cafe377682d35fd0c82cb816bf0a0fb0c0100/generate_build_files.sh#L112-L115). However, neither of those are specific to rules_nixpkgs, so it seems out of scope for this set of examples.

Reply:
I've added a Nix-only example in cc_with_deps.

Reply:
Oh, sorry, misread the request. I thought this was about something depending on the program. This makes sense, thanks @AlexeiDrake for adding it.